pypeit/images/pypeitimage.py

```diff
@@ -168,6 +170,12 @@ def from_pypeitimage(cls, pypeitImage):
         # Return
         return slf
 
+    # JFH This API or the documentation should make it clear what is actually required to perform operations with these
```

Comment:
Isn't it provided in the `datamodel` ?

Reply:
I think you should make the init for the class different from the datamodels and make the parameters required to create an image required parameters and the rest should be optional. That is a clearer API. I get that the datamodels needs all to be optional, but you can always make the object class separate from the datamodel class.

Reply:
I agree we should make this more clear.  The instantiation of the `DataContainer` does not require all of the init arguments to be optional.  See, e.g., `EdgeTraceSet`.